twitterpuller/src/main/java/de/brandwatch/minianalytics/twitterpuller/TwitterPullerApplication.java

```diff
@@ -15,12 +17,22 @@ public static void main(String[] args) {
         SpringApplication.run(TwitterPullerApplication.class, args);
     }
 
+    private final TwitterPullerStatusListener listener;
+
     @Autowired
-    private Producer producer;
+    public TwitterPullerApplication(TwitterPullerStatusListener listener) {
```

Comment:
No need to Autowire this. When you create a Bean with a param as you do below with the listener, it's automatically assumed to be a Bean as well and hence Autowired.

Also, you are calling the Bean method instead of using the Bean itself. I'd actually suggest to move away from using `PostConstruct` with a blocking method like your stream, as it will prevent your app to fully start. It will forever be stuck in the post construct phase.

I'd recommend just this:

```java
@Bean
public TwitterStream twitterStream(TwitterPullerStatusListener listener) {
    TwitterStream twitterStream = TwitterStreamFactory.getSingleton();
    twitterStream.addListener(listener);
    return twitterStream;
}


@Bean(initMethod = "sendTweetsToKafka")
public TweetProducer tweetProducer(TwitterStream twitterStream){
    return new TweetProducer(twitterStream);
}
```
together with a class `TweetProducer`:

```java
package de.brandwatch.minianalytics.twitterpuller.twitter;

import twitter4j.TwitterStream;

public class TweetProducer {

    private TwitterStream twitterStream;

    public TweetProducer(TwitterStream twitterStream) {
        this.twitterStream = twitterStream;
    }

    public void sendTweetsToKafka() {
        twitterStream.sample("de");
    }
}
```

I haven't tried it, but this should create the new `TweetProducer` as a bean and call the init method (`sendTweetsToKafka`) without  blocking anything.